app/controllers/concerns/locale.rb

```diff
@@ -0,0 +1,51 @@
+module Concerns::Locale
+  extend ActiveSupport::Concern
+
+  included do
+    before_filter :set_locale
+  end
+
+  def explicitly_requested_language
+    params[:locale]
+  end
+
+  def user_settings_language
+    current_user.locale if current_user
```

Comment:
`.try()`?

Reply:
:) Since we're using Ruby 2.3 anyway, why not switch to `current_user&.locale`?

Reply:
👍